api/app/dependency.py

```diff
@@ -0,0 +1,50 @@
+""" Authentication module for validating access tokens generated by Keycloak
+"""
+import logging
+from fastapi import Depends, HTTPException, status, Request
+from fastapi.security import OAuth2PasswordBearer
+import jwt
+from jwt import InvalidTokenError
+from app import config
+from app.db.crud.api_access_audits import create_api_access_audit_log
+
+logger = logging.getLogger(__name__)
+
+# Parse request header and pass the token
+oauth2_scheme = OAuth2PasswordBearer(tokenUrl="/token")
+
+
+async def authenticate(token: str = Depends(oauth2_scheme)):
```

Comment:
It's not logging un-authenticated attempts, since OAuth2PasswordBearer raises a 401 exception. You'll have to wrap the call that's happening in oauth2_scheme. Might be a good idea to add a test case for this scenario - it's really easy to miss.

Reply:
Oof thanks

Reply:
Apparently it doesn't actually raise a 401 exception: https://fastapi.tiangolo.com/tutorial/security/first-steps/#what-it-does

I think writing a test will flush out what needs to be changed here.

Reply:
See source code: https://github.com/tiangolo/fastapi/blob/master/fastapi/security/oauth2.py#L157

FastAPI generally follows the pattern of raising exception for HTTP error codes - that then get transformed into the appropriate response.

Reply:
Huh, good example of code not lining up with documentation...